### PR TITLE
Manta signer

### PR DESCRIPTION
**Pull request by Kevingislason**

### CODE REVIEW

**Comment by stechu** [comment]
Can you add a short description (possibly in bullet points) covers the design of the changes?

**Review by stechu — CHANGES_REQUESTED** [review]
LGTM in general. Have some minor comments here.
Also, please add a comment string for each function (at least all public ones). The comment string starts from `///`

File: src/signer/hd_wallet.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	1
+} else {
+	611
+};
+
+const DEFAULT_ACCOUNT_ID: u32 = 0;
+
+const EXTERNAL_CHAIN_ID: u32 = 0;
+const INTERNAL_CHAIN_ID: u32 = 1;
+
```
[the remark this answers is not in the document]
Comment (Kevingislason) [reply src/signer/hd_wallet.rs:40]: Looks consistent to me, not sure what you mean. What am I missing?

File: src/signer/params.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	pub asset_1_shard: Vec<[u8; 32]>,
+	pub asset_2_shard: Vec<[u8; 32]>,
+	pub change_path: DerivationPathString,
+	pub reclaim_value: AssetBalance,
+}
+
+#[derive(Debug, Clone, Encode, Decode, Default, PartialEq)]
+pub struct GeneratePrivateTransferDataParams {
+	pub asset_id: AssetId,
+	pub asset_1_value: AssetBalance,
```
[the remark this answers is not in the document]
Comment (Kevingislason) [reply src/signer/params.rs]: Yeah, `asset_1_*` is definitely vague, good call. Went with `sender_asset_1_*`, I think that is maximally unambiguous.

File: src/signer/payload_gen.rs [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+use ark_std::rand::{CryptoRng, RngCore};
+use codec::{Decode, Encode};
+use manta_asset::{AssetBalance, AssetId, MantaAsset, MantaSecretKey, Process, Sampling, UTXO};
+use manta_crypto::{CommitmentParam, HashParam, MantaSerDes, COMMIT_PARAM, HASH_PARAM};
+use manta_data::{BuildMetadata, MintData, PrivateTransferData, ReclaimData};
+use std::str::FromStr;
+use tiny_hderive::{bip32::ExtendedPrivKey, bip44::DerivationPath};
+
+#[derive(Debug, Clone, Encode, Decode, Default, PartialEq)]
+pub struct MantaUIAsset {
```
[the remark this answers is not in the document]
Comment (Kevingislason) [reply src/signer/payload_gen.rs:34]: I agree that the current name should be changed. But future wallet designs might store different asset data (e.g. secret keys instead of keypaths).

A bit wordy, but how do you like `MantaSignerInputAsset`?

**Comment by bhgomes** [comment]
Closing this in favor of #27.
